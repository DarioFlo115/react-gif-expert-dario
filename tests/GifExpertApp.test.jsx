import { fireEvent, getAllByText, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('Pruebas en GifExpertApp', () => { 
    
    test('debe de hacer match con el snapshot', () => { 
        
        const {container} = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
     });

     test('debe de agregar una nueva categoria cuando se envia el formulario', () => { 
        
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const inputValue = 'Saitama';

        fireEvent.input(input, {target:{value: inputValue}});

        fireEvent.submit(input);

        expect(screen.getByText(inputValue)).toBeTruthy();

      });

      test('debe de no agregar una categoria si esta ya existe', () => { 
        
        render(<GifExpertApp/>);

        const category = 'One Punch';
        const input = screen.getByRole('textbox');

        fireEvent.input(input, {target: {value: category}});
        fireEvent.submit(input);

        const items = screen.getAllByText(category);

        expect(items.length).toBe(1);



       });

        test('debe de renderizar las categorias que se agregen con el tiempo correctamente', () => { 

            render(<GifExpertApp/>);

            const form = screen.getByRole('form');
            const input = screen.getByRole('textbox');

            fireEvent.input(input, {target:{value:'Naruto'}});
            fireEvent.submit(form);
            fireEvent.input(input, {target:{value:'Dragon Ball'}});
            fireEvent.submit(form);

            const items = screen.getAllByRole('heading', {level: 3});

            expect(items.length).toBe(3);
            expect(screen.getByText('One Punch')).toBeTruthy();
            expect(screen.getByText('Naruto')).toBeTruthy();
            expect(screen.getByText('Dragon Ball')).toBeTruthy();

        });

 });