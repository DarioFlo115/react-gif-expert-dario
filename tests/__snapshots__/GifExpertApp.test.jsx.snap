// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en GifExpertApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
